File: 30-06-2025/p594.js
var findLHS = function(nums) {
    let freq = new Map();
    for (let num of nums) {
        freq.set(num, (freq.get(num) || 0) + 1);
    }

    let ans = 0;
    for (let [key, val] of freq.entries()) {
        if (freq.has(key + 1)) {
            ans = Math.max(ans, val + freq.get(key + 1));
        }
    }
    return ans;
};

File: 30-06-2025/p594.ex
defmodule Solution do
  def find_lhs(nums) do
    counts = Enum.frequencies(nums)

    Enum.reduce(counts, 0, fn {num, count}, max_length ->
      if Map.has_key?(counts, num + 1) do
        max(max_length, count + Map.fetch!(counts, num + 1))
      else
        max_length
      end
    end)
  end
end